Terminal.Gui/Input/Bindings.cs

```diff
@@ -0,0 +1,150 @@
+﻿#nullable enable
+using System;
+
+namespace Terminal.Gui;
+
+/// <summary>
+///     Abstract base class for <see cref="KeyBindings"/> and <see cref="MouseBindings"/>.
+/// </summary>
+/// <typeparam name="TEvent">The type of the event (e.g. <see cref="Key"/> or <see cref="MouseEventArgs"/>).</typeparam>
+/// <typeparam name="TBinding">The binding type (e.g. <see cref="KeyBinding"/>).</typeparam>
+public abstract class Bindings<TEvent, TBinding>  where TBinding : IInputBinding, new() where TEvent : notnull
+{
+    /// <summary>
+    ///     The bindings.
+    /// </summary>
+    protected readonly Dictionary<TEvent, TBinding> _bindings;
+
+    private readonly Func<Command [], TEvent, TBinding> _constructBinding;
+
+    /// <summary>
+    ///     Initializes a new instance.
+    /// </summary>
+    /// <param name="constructBinding"></param>
+    /// <param name="equalityComparer"></param>
+    protected Bindings (Func<Command [], TEvent, TBinding> constructBinding, IEqualityComparer<TEvent> equalityComparer)
+    {
+        _constructBinding = constructBinding;
+        _bindings = new (equalityComparer);
+    }
+
+    /// <summary>Adds a <see cref="TEvent"/> bound to <see cref="TBinding"/> to the collection.</summary>
```

Comment:
I think the correct syntax for this is `<typeref name=` (or similar).  Not `<see`.